### ISSUE TITLE
ALSA only setup crashes repeatedly by calls to Pulseaudio functions

### ISSUE THREAD

**Issue by NP-Hardass** [muammar/mkchromecast#157]
I decided to check out your project because it seemed really cool.  Unfortunately, on an ALSA only system, despite the recent effort to add ALSA support, the program will repeatedly crash because it still attempts to make calls to pulseaudio programs.

From what I can tell, this boils mostly down to calls to create_sink, check_sink, and remove_sink that are conditioned on platform, instead of something like backend.

For example,  https://github.com/muammar/mkchromecast/blob/master/mkchromecast/tray_threading.py#L96
```
        if platform == 'Linux':
            # We create the sink only if it is not available
            if check_sink() is False:
                create_sink()
```
In this specific instance, pulseaudio sinks should only be created if backend == 'parec', I believe, rather than checking platform, since it supports multiple backends.

Assuming that is the best course of action, when refactoring, it might be wise to check all 'platform' conditions (whether it be a darwin or linux check) and change them to 'backend' ones so that there is complete flexibility to use any backend on any (supported) platform without assuming a particular backend is synonymous with platform.

Thanks for the cool project.  I'm looking forward to being able to try it out in depth when the issue has been resolved.

**Comment by muammar** [comment]
Would you try to run mkchromecast from the devel branch? I think that the minor change referenced in c7de064 should fix the problem. Thanks for reporting this. I don't have a linux installation with pure ALSA right now.

**Comment by NP-Hardass** [comment]
I can certainly try that specific solution.   However, there are quite a few other *_sink calls throughout the program.  For instance, reset audio calls remove_sink, (and there are a couple other *_sink) references throughout that are likely to cause issues later.

I still think the better solution is to migrate everything from platform checks to backend ones though.  Shouldn't a linux user be able to use the node.js backend too, for instance, or the mac user able to use the ffmpeg backend?

**Comment by muammar** [comment]
Backends checks only I don't think are going to be too feasible. macOS is also using ffmpeg and avconv. Besides, when using ffmpeg for audio, you still need to create a sink to cast using pulseaudio. Therefore, a backend only check would be insufficient. Does it make sense?

**Comment by NP-Hardass** [comment]
Yeah, I think I understand.  I'll see where else I run into problems besides the reset audio call to remove_sink

**Comment by muammar** [comment]
Thanks! I also would be more than happy to accept PR if you are in the mood :).

**Comment by NP-Hardass** [comment]
Sure I can certainly try, but make no promises on account of lack of familiarity with the code/skill :P
